config/state.ts

```diff
@@ -0,0 +1,87 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { i18n } from '../utils/lang';
+import { STATE_FILE_PATH } from '../constants/config';
+
+const i18nKey = 'config.state';
+
+interface CLIState {
+  mcpTotalToolCalls: number;
+}
+
+const DEFAULT_STATE: CLIState = {
+  mcpTotalToolCalls: 0,
+};
+
+function ensureCLIDirectory(): void {
+  try {
+    const dir = path.dirname(STATE_FILE_PATH);
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir, { recursive: true });
+    }
+  } catch (error) {
+    throw new Error(
+      i18n(`${i18nKey}.ensureCLIDirectory.errors.cannotCreateDirectory`, {
+        error: error instanceof Error ? error.message : String(error),
+      })
+    );
+  }
+}
+
+export function getStateValue<K extends keyof CLIState>(key: K): CLIState[K] {
+  ensureCLIDirectory();
+
+  try {
+    if (fs.existsSync(STATE_FILE_PATH)) {
+      const data = fs.readFileSync(STATE_FILE_PATH, 'utf-8');
+      const state = JSON.parse(data) as CLIState;
+      return state[key];
+    }
+  } catch (error) {
+    throw new Error(
+      i18n(`${i18nKey}.getStateValue.errors.errorReading`, {
+        error: error instanceof Error ? error.message : String(error),
+      })
+    );
+  }
+
+  return DEFAULT_STATE[key];
+}
+
+export function setStateValue<K extends keyof CLIState>(
+  key: K,
+  value: CLIState[K]
+): boolean {
```

Comment:
Since this will throw an error rather than return false if something goes wrong, can probably just have it return `void` instead